app/models/user.rb

```diff
@@ -125,6 +126,19 @@ def set_avatar_from_twitter
   rescue
   end
 
+  # Extract the information that is available after OmniAuth authentication.
+  def self.from_omniauth(auth, params)
+    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
+      user.provider = auth.provider
+      user.uid = auth.uid
+      user.first_name = auth.info.name.split(" ").first
+      user.last_name = auth.info.name.split(" ").last
+      user.email = auth.info.email.downcase
+      user.password = Devise.friendly_token[0,20]
+      user.role = params["role"]
```

Comment:
Let's also set the image for the user: `user.remote_avatar_url = auth.info.image` – this will take the image URL from Facebook and set it on their avatar.